### ISSUE TITLE
Use error_messages in forms to make it easier to override the error messages in subclasses

### ISSUE THREAD

**Issue by YAmikep** [pennersr/django-allauth#653]
What about using error codes and an `error_messages` attribute in a Form to make it easier to override the error messages?

For example, in ChangePasswordForm:

```
class ChangePasswordForm(UserForm):
    # ...
    def clean_oldpassword(self):
        if not self.user.check_password(self.cleaned_data.get("oldpassword")):
            # Need to override the whole method just to change the error message
            raise forms.ValidationError(_("Please type your current"  
                                          " password."))
        return self.cleaned_data["oldpassword"]
```

That could become:

```
class ChangePasswordForm(UserForm):
    #...    
    error_messages = {
            # this can be overridden by the subclasses to change the error messages
            'password_incorrect': _("THE PASSWORD IS INCORRECT."), 
        }

    def clean_oldpassword(self):
        if not self.user.check_password(self.cleaned_data.get("oldpassword")):
            raise forms.ValidationError(
                # use the error_messages attribute
                self.error_messages['password_incorrect'],
                code='password_incorrect',
            )
        return self.cleaned_data["oldpassword"]
```

**Comment by bitcity** [comment]
My 2 cents : I'd like to customize a number of error messages but overriding a class just for that seems like too much trouble. Is there an easier way ?

**Comment by bitcity** [comment]
+1 for using templates to override error messages

**Comment by dogukankotan** [comment]
+1

**Comment by boscharun** [comment]
+1 for having template to override error messages.

**Comment by cemonel** [comment]
+1 for having template to override error messages.

**Comment by derek-adair** [comment]
JEEBUS people just hit the thumbs up and dont spam...... what is this 2010?

----------------
Close this and open a feature to convert error messages to templates..... I guess? Personally, when I want custom forms I've found it's best to inherit or flat-out override them.  If it was **MY** project i'd close this w/ PR welcome, there "could" be an easier way to extend these templates but... meh.  Lets focus on more pressing matters.

**Comment by pennersr** [comment]
I think we just need to move all over to the adapter:

```
class DefaultAccountAdapter(object):
    error_messages = {
        "username_blacklisted": _(
            "Username can not be used. Please use other username."
        ),
```

Some are already there but it has not been consistently be applied throughout the whole project.

**Comment by derek-adair** [comment]
A nice low level ticket for anyone looking to get some street cred.

**Comment by pennersr** [comment]
Fixed via 481928bd

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/653
